public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/contactPoint/ContactPointSelector.tsx

```diff
@@ -1,75 +1,198 @@
-import { css } from '@emotion/css';
-import React from 'react';
+import { css, cx } from '@emotion/css';
+import { BaseQueryFn, QueryDefinition } from '@reduxjs/toolkit/dist/query';
+import { QueryActionCreatorResult } from '@reduxjs/toolkit/dist/query/core/buildInitiate';
+import React, { useCallback, useEffect, useState } from 'react';
 import { useFormContext } from 'react-hook-form';
 
 import { GrafanaTheme2, SelectableValue } from '@grafana/data';
-import { ActionMeta, Field, FieldValidationMessage, InputControl, Select, Stack, useStyles2 } from '@grafana/ui';
+import { BackendSrvRequest } from '@grafana/runtime';
+import {
+  ActionMeta,
+  Field,
+  FieldValidationMessage,
+  IconButton,
+  InputControl,
+  Select,
+  Stack,
+  TextLink,
+  useStyles2,
+} from '@grafana/ui';
 import { RuleFormValues } from 'app/features/alerting/unified/types/rule-form';
+import { createUrl } from 'app/features/alerting/unified/utils/url';
+import { AlertManagerCortexConfig } from 'app/plugins/datasource/alertmanager/types';
 
-import { ContactPointReceiverSummary } from '../../../../contact-points/ContactPoints';
 import { ContactPointWithMetadata } from '../../../../contact-points/utils';
 
 export interface ContactPointSelectorProps {
   alertManager: string;
-  contactPoints: ContactPointWithMetadata[];
+  options: Array<{
+    label: string;
+    value: ContactPointWithMetadata;
+    description: React.JSX.Element;
+  }>;
   onSelectContactPoint: (contactPoint?: ContactPointWithMetadata) => void;
+  refetchReceivers: () => QueryActionCreatorResult<
```

Comment:
The return value of this function doesn't seem to be used in this component.
How about renaming it to `onRefreshClick` and typing as a simple `() => Promise<void>`?

Reply:
`() => Promise<void>` is incompatible with the refetch from `RTKQ`

Reply:
`() => Promise<unknown>` works 🎉